src/cpu.rs

```diff
@@ -343,18 +403,450 @@ impl Cpu {
         // TODO
         0
     }
+
+    fn ana(&mut self, val: u8) {
+        self.and(val)
+    }
+
+    fn ani(&mut self, val: u8) {
+        self.and(val)
+    }
+
+    fn xra(&mut self, val: u8) {
+        self.xor(val)
+    }
+
+    fn xri(&mut self, val: u8) {
+        self.xor(val)
+    }
+
+    fn ora(&mut self, val: u8) {
+        self.or(val)
+    }
+
+    fn ori(&mut self, val: u8) {
+        self.or(val)
+    }
+
+    fn cmp(&mut self, val: u8) {
+        self.compare(val)
+    }
+
+    fn cpi(&mut self, val: u8) {
+        self.compare(val)
+    }
+
+    fn and(&mut self, val: u8) {
+        self.registers.a = self.registers.a & val;
+
+        self.condition_codes.reset_carry();
+        self.condition_codes.set_sign(self.registers.a);
+        self.condition_codes.set_zero(self.registers.a);
+        self.condition_codes.set_parity(self.registers.a);
+    }
+
+    fn xor(&mut self, val: u8) {
+        self.registers.a = self.registers.a ^ val;
+
+        self.condition_codes.reset_carry();
+        self.condition_codes.set_sign(self.registers.a);
+        self.condition_codes.set_zero(self.registers.a);
+        self.condition_codes.set_parity(self.registers.a);
+    }
+
+    fn or(&mut self, val: u8) {
```

Comment:
which instruction is this one for?

Reply:
This is for ora and ori - they both pass in a value to this. Many of my sets of functions are like this like and, jump, compare, etc. as there are different types of the instructions.

Reply:
Ahh, coo coo.